src/components/shared/Entity/Actions/SchemaEvolution.tsx

```diff
@@ -0,0 +1,169 @@
+
+        jobStatusPoller(
+            supabaseClient
+                .from(TABLES.EVOLUTIONS)
+                .select(`${JOB_STATUS_COLUMNS}, collections, draft_id`)
+                .match({
+                    draft_id: draftIdVal,
+                    logs_token: logTokenVal,
+                }),
+            async (payload: any) => {
+                // Update the state so it knows that things came back correctly
+                setFormState({
+                    status: FormStatus.SCHEMA_EVOLVED,
+                });
+
+                // Treat these like discovered collections. This will mean the UI
+                //  will go fetch all the new collections, specs, configs and store
+                //  them locally and present them
+                await storeDiscoveredCollections(
+                    payload.draft_id,
+                    entityType,
+                    onFailure
+                );
+
+                // Now that the collections are updated we can clear out the list so
+                //  the error section is no longer displayed.
+                setIncompatibleCollections([]);
+            },
+            async (payload: any) => {
+                if (payload.error === JOB_STATUS_POLLER_ERROR) {
+                    jobFailed(DEFAULT_POLLER_ERROR);
+                } else {
+                    jobFailed({
+                        ...DEFAULT_POLLER_ERROR,
+                        message: `entityEvolution.serverUnreachable`,
+                    });
+                }
+            }
+        );
+    };
+
+    const save = async (event: React.MouseEvent<HTMLElement>) => {
+        event.preventDefault();
+
+        updateFormStatus(FormStatus.SCHEMA_EVOLVING);
+
+        if (draftId) {
+            // eslint-disable-next-line @typescript-eslint/no-unnecessary-condition
+            if (incompatibleCollections && incompatibleCollections.length > 0) {
+                const response = await createEvolution(
+                    draftId,
+                    incompatibleCollections.map(
+                        (incompatibleCollection) =>
+                            incompatibleCollection.collection
+                    )
+                );
+                if (response.error) {
+                    onFailure({
+                        error: {
+                            title: `entityEvolution.failure.errorTitle`,
+                            error: response.error,
+                        },
+                    });
+                } else {
+                    waitForEvolutionToFinish(
+                        response.data[0].logs_token,
+                        draftId
+                    );
+                    setFormState({
+                        logToken: response.data[0].logs_token,
+                        showLogs: false,
+                    });
+                }
+            }
+        } else {
+            logRocketEvent('Schema:Evolution:Missing draftId');
+            onFailure({
```

Comment:
`Missing Draft Id` seems like a confusing error message to result from this. It kinda seems like we'd want to treat this as some type of "internal error". WDYT? I'm assuming this would be very uncommon in any case.

Reply:
It has never happened as far as we know and also not really sure how this would happen in the real world but we just put it there cause it _could_ happen.